server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -934,7 +940,11 @@ private long balanceTablets() {
         synchronized (balancedNotifier) {
           balancedNotifier.notifyAll();
         }
+        balancerMetrics.setMigratingCount(0);
+        balancerMetrics.setNeedMigrationCount(0);
       } else {
+        balancerMetrics.setMigratingCount(params.migrationsOut().size());
+        balancerMetrics.setNeedMigrationCount(migrations.size());
```

Comment:
I believe that the `migrations` variable in the Manager contains all of the in-flight migrations. The `params.migrationsOut` contains the tablets that need to be migrated from this call to the balancer. I think based on what I'm seeing here, it's double counting. I wonder if we just want the size of the `migrations` list as that will encompass newly added, newly removed, and migrations waiting to happen.

Reply:
If you want to capture the newly added migrations, then it might make sense to capture the migration completions (which are in the TabletGroupWatcher) and cancellations (which happen in a couple places in the Manager).

Reply:
So to complete the #4470 ask, just recording `migrations.size()` would be sufficient. 
`migrationsOut().size()` seems useful as a secondary metric for alerting if balancer progress stalls.

Doing some general tests and with a single balancer call, both of the currently implemented metrics come back with the same values.

I think keeping `migrations.size()` under the `migrations.needed` metric makes sense. 
`in-progress` wording may need to be tweaked a bit to better describe last set of migrations generated. 

For users, a possible alert condition would be `migrationsOut().size()` is 0 while `migrations.size()` is still greater than 0.